File: AccelerationToMulitcore/1.CPUOpenmp/00.ParallelExecution/step1/HelloWorld.c
#include <stdio.h>
#include <omp.h> 

int main(int argc, char* argv[])
{
  int nthreads = 4;

#pragma omp parallel
{
  int id = omp_get_thread_num();

  printf("Hello(%d)\n",id);
  printf("world(%d)\n",id);
}
  return 0;
}

File: AccelerationToMulitcore/1.CPUOpenmp/00.ParallelExecution/step1/HelloWorld.f90
! =========================
!  Hello world
! =========================
program helloWorld_omp_parallel

  USE omp_lib

  implicit none 
  INTEGER :: nthreads=4
  integer :: tid
  
  CALL OMP_SET_NUM_THREADS(nthreads)

  !$omp parallel
  tid = OMP_GET_THREAD_NUM()
  print *, "Hello World from thread = ", tid, " with ", OMP_GET_NUM_THREADS(), " threads" 
  !$omp end parallel

  PRINT *, "all done, with hopefully ",nthreads," threads"

end program helloWorld_omp_parallel
